### PR TITLE
Bump `uuid` and `dashmap` dependencies.

### PR DESCRIPTION
**Pull request by lseelenbinder**
What it says on the cover.

- Bumps `uuid` version to 1.0.0 (which may break downstream code due to mismatching versions, but doesn't break the API).
- Bumps `dashmap` for relevant speedups.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: scylla/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "scylla"
-version = "0.4.4"
+version = "0.5.0"
```
Comment (psarna) [inline scylla/Cargo.toml:3]: That bump will be done once we're ready to release, via our release script. Scylla is still in 0.x, so we have relaxed policy about specific version, and it's more likely we'll only release 0.5.0 after some new features arrive. Let's just drop this change from this patch.

**Review by psarna — COMMENTED** [review]
Looks good, but while we're bumping, let's bump to a minor release instead of a specific patch one.

Historically we specified everything down to patch versions (e.g. `uuid = "0.8.1"`), but it was a mistake on our end, and it's better to incrementally move to specifying only minor releases. So in this case, `uuuid = "1.0"` and `dashmap = "5.2"`

**Comment by lseelenbinder** [comment]
@psarna Changes made as requested. Thanks!

Probably want to squash-merge this one when the time comes. :smile:

**Comment by psarna** [comment]
Merged, thanks for contributing!

**Pull request merged**